src/main/java/edu/kit/compiler/semantic/NamespaceMapper.java

```diff
@@ -0,0 +1,42 @@
+package edu.kit.compiler.semantic;
+
+import edu.kit.compiler.data.ast_nodes.ClassNode;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class NamespaceMapper {
+
+    private final Map<ClassNode, ClassNamespace> symbolTableMap = new HashMap<>();
+
+    @AllArgsConstructor
+    @Data
+    public static class ClassNamespace {
+        private SymbolTable methodSymbols;
```

Comment:
In the AST we differentiate between static methods and dynamic methods, would it make sense to make that distinction here as well?

Reply:
We should definitively differentiate the two. The question is probably whether we want to add any static method to the name space or, instead, directly check whether it is the main method (which is the only semantically valid static method).

Reply:
I would just use the extra data structure even if it costs us a tiny amount of performance, since its nicer to read